### DISCUSSION TITLE
Is there anyway to communicate with Android's Chrome ?

### DISCUSSION THREAD

**Discussion by yuroyami** [joffrey-bion/chrome-devtools-kotlin#245] [Q&A]
First of all, thank you so much for this amazing repository. It does exactly what it says. I am taking it a step further and I wanted to access Chrome's UNIX-Domain Socket (which returns **_Permission Denied_** due to SELInux's Policies) since that's the only way to actually receive any data from the browser towards DevTools. 
Does any workaround come to mind ? This repository saves the majority of work and actually set up everything on one device. But yet, it would be really interesting to bridge communication between this library (and the app wrapping it) and Chrome's Android Browser. Android uses "sandboxing" scheme as far as I know.

**Comment by joffrey-bion** [comment]
Hi, thanks a lot for your feedback, I'm glad you find this library helpful 😊 

I'm not sure I'm getting your use case correctly. You are saying that you have an Android application that uses chrome-devtools-kotlin and wants to connect to the local Chrome application on the same device?

I don't know much about the chrome debugging options on Android unfortunately, but if chrome doesn't expose its debugger port, you won't be able to connect I guess. Maybe enabling USB debugging also interacts with the Chrome debugger flag in a way?

**Comment by JingMatrix** [comment]
You need to forward the Android Chrome socket for DevTools.
Here is an example in my project: https://github.com/JingMatrix/ChromeXt/tree/master/app/src/main/java/org/matrix/chromext/devtools
